### ISSUE TITLE
Improve Governor Guide

### ISSUE THREAD

**Issue by Heph789** [OpenZeppelin/openzeppelin-contracts#4619]
**🧐 Motivation**

As someone learning how to implement on-chain governance through OZ's Governor contract, I found the Governance in the guide to be missing important information and in some cases, outdated. I'd like to make it better!

**📝 Details**

Here are a few of the key missing areas I've found:
- There should be more details included about how to actually deploy the token contract. The example contract code given isn't actually suitable, since it doesn't mint any token upon creation.
- The sample governor contract uses the timelock controller, but there is no sample contract provided that extends GovernorTimelockControl, and there is not enough information for someone starting out to be able to infer how to do this properly
- The example code uses hardhat's extension of ether.js, but it does not mention hardhat in the guide. Someone using ether.js in a different environment would not be able to run the example code
- "ethers.utils.id" is now just "ethers.id" due to a new ethers version

Overall, I think it would be better if this guide were actually a walkthrough. "Deploy on-chain governance locally on hardhat!" That way there is an actual example that people can interact with locally. I'd love to hear yall's thoughts on this!

**Comment by frangio** [comment]
I agree with the bullet points, and I think I agree that a walkthrough would be nice. However, the current guide is also a bit of a conceptual introduction and overview so that doesn't exactly fit in a walkthrough.

Feel free to submit a PR for the bullet points.

**Comment by Heph789** [comment]
A couple of questions and comments:
Addressing the first two bullet points would require changing the sample code provided in the guide, but that code is behind an api somewhere (referenced like this: ```link:api:example$governance/MyToken.sol[]```) and I'm not sure how/if I can access that.

For the last third bullet point, should I just include the fact that hardhat is being used, and add a note about which version is being used, and which function is a hardhat injection? Or should I try to change it so it works solely with ethers.js? Solely using ethers.js might increase the complexity of the code a fair amount.

**Comment by ernestognw** [comment]
> but that code is behind an api somewhere (referenced like this: link:api:example$governance/MyToken.sol[]) 

You can find them under the [`mocks/docs/**` folder.](https://github.com/OpenZeppelin/openzeppelin-contracts/tree/1169bb1e513e78dc4d8424689625d19b8a832fa6/contracts/mocks/docs) The API reference is the route after this folder.

Regarding ethersjs, I think mentioning briefly at the beginning as a prerequisite will be fine. Additionally, feel free to add comments to the code snippets if you think they're useful.
